src/core/utils/pick.ts

```diff
@@ -17,16 +17,19 @@
  * under the License.
  */
 
-export function pick<T extends { [k: string]: any }, K extends keyof T>(
+export function pick<T extends Record<string, unknown>, K extends keyof T>(
   obj: T,
   keys: K[]
 ): Pick<T, K> {
   const newObj = keys.reduce(
-    (acc, val) => {
-      acc[val] = obj[val];
+    (acc, key) => {
+      if (key in obj) {
```

Comment:
should we allow to pick from prototype chain? if not it's better to use `hasOwnProperty` here

Reply:
Yeah, it just makes sense to use `Object.keys` here (will do) and later on just downgrade this "generic" utility to more specialized one and place it where it's actually used.

Reply:
Ah, didn't realize we're inside of `reduce` already, yeah will use `hasOwnProperty`.